client/web/src/repo/blob/own/MakeOwnerButton.tsx

```diff
@@ -0,0 +1,54 @@
+import { mdiLoading, mdiPlus } from '@mdi/js'
+
+import { useMutation } from '@sourcegraph/http-client'
+import { Button, Icon, Tooltip } from '@sourcegraph/wildcard'
+
+import { AssignOwnerResult, AssignOwnerVariables } from '../../../graphql-operations'
+
+import { ASSIGN_OWNER } from './grapqlQueries'
+
+export interface MakeOwnerButtonProps {
+    onSuccess: () => Promise
+    onError: (e: Error) => void
+    repoId: string
+    path: string
+    userId?: string // TODO make optional, user ID is not always known
+}
+
+export const MakeOwnerButton: React.FC<MakeOwnerButtonProps> = ({ onSuccess, onError, path, userId, repoId }) => {
+    const [requestAssignOwner, { loading }] = useMutation<AssignOwnerResult, AssignOwnerVariables>(ASSIGN_OWNER, {})
+
+    const assignOwner = async () => {
+        if (userId !== undefined) {
+            const result = await requestAssignOwner({
+                variables: {
+                    input: {
+                        absolutePath: path,
+                        assignedOwnerID: userId,
+                        repoID: repoId,
+                    },
+                },
+            })
+            if (result.errors) {
+                onError(new Error('Failed to make owner.'))
+            } else {
+                await onSuccess()
+            }
+        }
+    }
+
+    const tooltipContent =
+        userId === undefined
+            ? 'Only ownership entries that are recognized as Sourcegraph users can be assigned ownership.'
+            : null
+
+    return (
+        // TODO: Add tooltip
+        <Tooltip content={tooltipContent}>
+            <Button variant={'primary'} outline={true} size={'sm'} disabled={userId === undefined}>
```

Comment:
Can we simplify it like this?
`useMutation` hook can pretty much handle all this stuff

```suggestion
    const [assignOwner, { loading }] = useMutation<AssignOwnerResult, AssignOwnerVariables>(ASSIGN_OWNER, {
        variables: {
            input: {
                absolutePath: path,
                assignedOwnerID: userId,
                repoID: repoId,
            },
        },
        onError: () => onError(new Error('Failed to make owner.')),
        onCompleted: onSuccess,
    })

    const tooltipContent =
        userId === undefined
            ? 'Only ownership entries that are recognized as Sourcegraph users can be assigned ownership.'
            : null

    return (
        <Tooltip content={tooltipContent}>
            <Button variant={'primary'} outline={true} size={'sm'} onClick={() => assignOwner()} disabled={userId === undefined}>
```